runtime/common/src/account_conversion.rs

```diff
@@ -18,6 +18,14 @@ use sp_core::{Get, H160};
 use sp_runtime::traits::Convert;
 use xcm::*;
 
+pub fn to_evm_address(account: &AccountId) -> H160 {
+	// EVM AccountId20 addresses are right-padded to 32 bytes
+	let mut bytes20 = [0; 20];
+	let account_32: [u8; 32] = account.clone().into();
+	bytes20.copy_from_slice(&account_32);
+	H160::from(bytes20)
+}
+
```

Comment:
Not sure exactly why, but `PrecompileSetBuilder::user_addresses()` returns an `AccountId` instead of `H160` (internally, it transforms `H160` into `AccountId`). 

So, we need this method to go back from `AccountId` to `H160` in `chain_spec.rs`. 

I really think it is taking more responsibility than it should, but I have no choice right now with the API they offer. Maybe I can open an issue regarding this in Moonbeam.

Reply:
Ok, I guess `H160::from_slice(&sender.as_ref()[0..20])` should be extracted to a converter.

Reply:
I've been juggling traits in this commit: [0fb227c](https://github.com/centrifuge/centrifuge-chain/pull/1731/commits/0fb227c5020d7f247d75e6b11b1b067e84507c33), and I've got to remove the conversion, so there is no issue (at least by now).